ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -133,7 +133,7 @@ internal constructor(
 
   // Tracks the current task ID to compute the position in the list of tasks for the current job.
   var currentTaskId: StateFlow<String> =
-    savedStateHandle.getStateFlow(TASK_POSITION_ID, tasks.first().id)
+    savedStateHandle.getStateFlow(TASK_POSITION_ID, tasks.firstOrNull()?.id ?: "")
```

Comment:
Do we need to update anything else to check if taskId is empty?

Reply:
That is already handled. We use it to check current position and if it's empty, we fallback to first position